### ISSUE TITLE
[Bug] offset_polygon function error during validation step

### ISSUE THREAD

**Issue by skyandd** [open-mmlab/mmocr#1688]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmocr/issues) and [Discussions](https://github.com/open-mmlab/mmocr/discussions) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (0.x)](https://github.com/open-mmlab/mmocr) or [latest version (1.x)](https://github.com/open-mmlab/mmocr/tree/dev-1.x).

### Task

I have modified the scripts/configs, or I'm working on my own tasks/models/datasets.

### Branch

1.x branch https://github.com/open-mmlab/mmocr/tree/dev-1.x

### Environment

Result `python mmocr/utils/collect_env.py` below:
[my_env.txt](https://github.com/open-mmlab/mmocr/files/10467009/my_env.txt)

I was following installing guide during the setting environment. 
["installing guide"](https://mmocr.readthedocs.io/en/dev-1.x/get_started/install.html)

Also, I installed `ipykernel` library for setting Jupiter notebook env

### Reproduces the problem - code sample

```
import pyclipper
import numpy as np

poly = [[386, 1179],
        [401, 1179],
        [400, 1178],
        [399, 1179],
        [398, 1178],
        [393, 1178],
        [392, 1179],
        [391, 1178],
        [389, 1178],
        [388, 1179],
        [387, 1178]]

distance = 0.49529160046840287

pco = pyclipper.PyclipperOffset()
pco.AddPath(poly, pyclipper.JT_ROUND, pyclipper.ET_CLOSEDPOLYGON)
# Returned result will be in type of int32, convert it back to float32
# following MMOCR's convention
res = pco.Execute(distance)
#ERROR
np.array(res)
```

### Reproduces the problem - command or script

Full console output during training + error
[training_outputs.txt](https://github.com/open-mmlab/mmocr/files/10467173/training_outputs.txt)

### Reproduces the problem - error message

```
/home/addudkin/text_detection/venv/lib/python3.8/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
/home/addudkin/text_detection/venv/lib/python3.8/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
/home/addudkin/text_detection/venv/lib/python3.8/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
Traceback (most recent call last):
  File "tools/train.py", line 131, in <module>
    main()
  File "tools/train.py", line 127, in main
    runner.train()
  File "/home/addudkin/text_detection/venv/lib/python3.8/site-packages/mmengine/runner/runner.py", line 1686, in train
    model = self.train_loop.run()  # type: ignore
  File "/home/addudkin/text_detection/venv/lib/python3.8/site-packages/mmengine/runner/loops.py", line 96, in run
    self.runner.val_loop.run()
  File "/home/addudkin/text_detection/venv/lib/python3.8/site-packages/mmengine/runner/loops.py", line 347, in run
    self.run_iter(idx, data_batch)
  File "/home/addudkin/text_detection/venv/lib/python3.8/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/home/addudkin/text_detection/venv/lib/python3.8/site-packages/mmengine/runner/loops.py", line 367, in run_iter
    outputs = self.runner.model.val_step(data_batch)
  File "/home/addudkin/text_detection/venv/lib/python3.8/site-packages/mmengine/model/base_model/base_model.py", line 133,in val_step
    return self._run_forward(data, mode='predict')  # type: ignore
  File "/home/addudkin/text_detection/venv/lib/python3.8/site-packages/mmengine/model/base_model/base_model.py", line 326,in _run_forward
    results = self(**data, mode=mode)
  File "/home/addudkin/text_detection/venv/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/addudkin/text_detection/mmocr/mmocr/models/textdet/detectors/base.py", line 74, in forward
    return self.predict(inputs, data_samples)
  File "/home/addudkin/text_detection/mmocr/mmocr/models/textdet/detectors/single_stage_text_detector.py", line 107, in predict
    return self.det_head.predict(x, data_samples)
  File "/home/addudkin/text_detection/mmocr/mmocr/models/textdet/heads/db_head.py", line 183, in predict
    predictions = self.postprocessor(outs, batch_data_samples)
  File "/home/addudkin/text_detection/mmocr/mmocr/models/textdet/postprocessors/base.py", line 64, in __call__
    results = list(map(process_single, pred_results, data_samples))
  File "/home/addudkin/text_detection/mmocr/mmocr/models/textdet/postprocessors/base.py", line 79, in _process_single
    results = self.get_text_instances(pred_result, data_sample, **kwargs)
  File "/home/addudkin/text_detection/mmocr/mmocr/models/textdet/postprocessors/db_postprocessor.py", line 102, in get_text_instances
    poly = self._unclip(poly_pts)
  File "/home/addudkin/text_detection/mmocr/mmocr/models/textdet/postprocessors/db_postprocessor.py", line 169, in _unclip
    return offset_polygon(poly_pts, distance)
  File "/home/addudkin/text_detection/mmocr/mmocr/utils/polygon_utils.py", line 335, in offset_polygon
    result = np.array(pco.Execute(distance))
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.
```

### Additional information

I started training DBnet++ for global text detectors on a custom dataset. The training of the model is going well, the model converges, and the visualization of the samples also showed the correct markup.

The error occurs at the time of the validation step in the `def offset_polygon` function. The fact is that in line 335
`result = np.array(pco.execute(distance))`
`/mmocr/mmocr/utils/polygon_utils.py` an array is formed with a list inside which the elements are of different lengths.

I was following installing guide during the setting environment. 
["installing guide"](https://mmocr.readthedocs.io/en/dev-1.x/get_started/install.html)

During the creation of the dataset, I was guided by the following [documentation](https://mmocr.readthedocs.io/en/dev-1.x/user_guides/dataset_prepare.html)

**Comment by jturner116** [comment]
I'm getting this bug also in a new environment, I think it is a regression

**Comment by gaotongxiao** [comment]
Thanks for the detailed information! It has been addressed in #1691 and already got fixed at `dev-1.x` branch. `1.x` branch will be updated as 1.0.0rc6 released.
